src/Polly.Extensions/Telemetry/EnrichmentUtil.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Collections.Generic;
+using Polly.Strategy;
+
+namespace Polly.Extensions.Telemetry;
+
+internal static class EnrichmentUtil
+{
+    public static void Enrich(
+        ref TagList tags,
+        List<Action<EnrichmentContext>> enrichers,
+        ResilienceContext resilienceContext,
+        Outcome? outcome,
+        IResilienceArguments? resilienceArguments)
+    {
+        if (enrichers.Count == 0)
+        {
+            return;
+        }
+
+        var context = EnrichmentContext.Get(resilienceContext, resilienceArguments, outcome);
+
+        foreach (var enricher in enrichers)
+        {
+            enricher(context);
+        }
+
+        foreach (var pair in context.Tags)
+        {
+            tags.Add(pair.Key, pair.Value);
+        }
```

Comment:
```suggestion
        if (context.Tags > 0)
        {
            foreach (var pair in context.Tags)
            {
                tags.Add(pair.Key, pair.Value);
            }
        }
```

Reply:
This seems artificial, once enricher is registered it will add a tag to the collection. Otherwise, there is no point registering it. 

So the expectation is that `context.Tags` is never empty when we have enrichers.